providers/fips/fipsprov.c

```diff
@@ -27,10 +27,14 @@
 #include "internal/property.h"
 #include "crypto/evp.h"
 #include "prov/implementations.h"
+#include "prov/provider_util.h"
 #include "prov/provider_ctx.h"
 #include "prov/providercommon.h"
 #include "selftest.h"
 
+#define ALGC(NAMES, FUNC, CHECK) { { NAMES, "fips=yes", FUNC }, CHECK }
+#define ALG(NAMES, FUNC) ALGC(NAMES, FUNC, NULL)
```

Comment:
Cool, this is workable with #10124.  All that's needed there is to add a checking function attribute for those who need it and let the perl nugget in the template do its work.